scripts/service-update-scripts/dcos-commons-merge.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+set -e
+
+STABLE_BRANCH="sdk-0.40"
+BETA_BRANCH="master"
+
+REPO_NAME="dcos-commons"
+REPO_URL="https://github.com/mesosphere/${REPO_NAME}.git"
+
+syntax() {
+    cat <<EOF
+This script copies docs for the specified framework from ${REPO_NAME} into this repository.
+Syntax:
+  $0 <[beta-]packagename> <version[-beta]> <packagetitle>
+Examples:
+  $0 beta-cassandra 2.1.0-3.0.6-beta Cassandra
+  $0 hdfs 2.1.0-4.1.0 HDFS
+EOF
+    exit 1
+}
+
+# Get values for version and directory variable
+
+package=$1
+if [ -z "$package" ]; then syntax; fi
+version=$2
+if [ -z "$version" ]; then syntax; fi
+title=$3
+if [ -z "$title" ]; then syntax; fi
+
+echo "------------------------------"
+echo " Merging $package from dcos-commons"
+echo "------------------------------"
+
+# Determine if the package is a beta or not
+if [[ "$package" == beta-* ]]; then
+    branch=$BETA_BRANCH
+    # Verify that version ends with '-beta'
+    if [[ "$version" != *-beta ]]; then
+        echo "Package $package requires a version ending in '-beta', but version was $version"
+        exit 1
+    fi
+else
+    branch=$STABLE_BRANCH
+    # Verify that the version doesn't end with '-beta'
+    if [[ "$version" == *-beta ]]; then
+        echo "Package $package requires a version that doesn't end in '-beta', but version was $version"
+        exit 1
+    fi
+fi
+
+# Go to dcos-docs-site root directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd ${SCRIPT_DIR}/../..
+pwd
+
+# Grab correct branch from repo
+rm -rf $REPO_NAME/
+git clone --depth 1 --branch $branch $REPO_URL
+
+# Omit any 'beta-' prefix from package name when getting dcos-commons path:
+framework=$(echo $package | sed 's/^beta-//g')
+input_framework_docs_dir=$REPO_NAME/frameworks/${framework}/docs
+
+output_package_dir=pages/services/${package}
+output_package_version_dir=${output_package_dir}/${version}
+if [ -d $output_package_version_dir ]; then
+   echo "Output directory already exists, remove first: $(pwd)/$output_package_version_dir"
+   exit 1
+fi
+
+# Before copying content, add 10 to the sort order of existing index files
+for i in $( ls ${output_package_dir}/*/index.md ); do
+    awk '/^menuWeight:/ {sub(/[[:digit:]]+$/,$NF+10)}1 {print}' $i > $i.tmp && mv $i.tmp $i
+done
```

Comment:
^-- SC2045: Iterating over ls output is fragile. Use globs
https://github.com/koalaman/shellcheck/wiki/SC2045